PresentData/PresentMonTraceConsumer.cpp

```diff
@@ -515,11 +547,15 @@ void PMTraceConsumer::HandleDxgkPropagatePresentHistoryEventArgs(EVENT_HEADER co
     if (eventIter->second->PresentMode == PresentMode::Composed_Composition_Atlas ||
         (eventIter->second->PresentMode == PresentMode::Composed_Flip && !eventIter->second->SeenWin32KEvents)) {
         mPresentsWaitingForDWM.emplace_back(eventIter->second);
+        eventIter->second->WaitingForDwmCompletion = true;
     }
 
     if (eventIter->second->PresentMode == PresentMode::Composed_Copy_GPU_GDI) {
         // Manipulate the map here
         // When DWM is ready to present, we'll query for the most recent blt targeting this window and take it out of the map
+
+        // TODO: what if there is already an element in mLastWindowPresent?

```

Comment:
Yes, there could've been one here. My intention when writing this was that any that got "lost" here should only remain in temporary structures that were manipulated by `CompletePresent`, and would be removed once one blt frame eventually completed -- that'd cause all the previous ones to also be completed and marked as Discarded, and cleaned up at that point.

Reply:
Awesome, thanks for the clarification! Forgot to remove this comment in the last revision, will address in the next one.